skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -638,4 +638,28 @@ static function onArticleSaveComplete( &$article, &$user, $text, $summary,
 		$wgMemc->delete( wfMemcKey( 'mOasisRecentRevisions2', $article->getTitle()->getArticleId() ) );
 		return true;
 	}
+
+	private function renderTemplateSubtitle() {
+		global $wgUser;
+		$templateType = $this->getTemplateType();
+		if ( $wgUser->isLoggedIn() ) {
+			$templateType .= $this->renderEditButton();
+			return $templateType;
+		} else {
+			if ( $templateType !== 'Unclassified' ) {
+				return $templateType;
+			}
+			return wfMessage( 'oasis-page-header-subtitle-template' )->escaped();
+		}
+
+	}
+
+	private function getTemplateType() {
+		// Mock for frontend work
+		return wfMessage( 'oasis-page-header-subtitle-template-unclassified' )->escaped();
+	}
+
+	private function renderEditButton() {
```

Comment:
I think that this name is misleading in the context of this Controller since it renders a button to edit Template Classification and not to perform a regular edit.